internal/command/views/modules.go

```diff
@@ -38,9 +39,34 @@ type ModulesHuman struct {
 var _ Modules = (*ModulesHuman)(nil)
 
 func (v *ModulesHuman) Display(manifest moduleref.Manifest) int {
+	if len(manifest.Records) == 0 {
+		v.view.streams.Println("No modules found in configuration.")
+		return 0
+	}
+	printRoot := treeprint.New()
+	populateTreeNode(printRoot, &moduleref.Record{
+		Children: manifest.Records,
+	})
+
+	v.view.streams.Println(fmt.Sprintf("\nModules declared by configuration:\n%s", printRoot.String()))
 	return 0
 }
 
+func populateTreeNode(tree treeprint.Tree, node *moduleref.Record) {
+	for _, childNode := range node.Children {
+		item := fmt.Sprintf("\"%s\"[%s]", childNode.Key, childNode.Source.String())
+		if childNode.Version != nil {
+			item += fmt.Sprintf(" %s", childNode.Version)
+			// Avoid rendering the version constraint if an exact version is given i.e. 'version = "1.2.3"'
+			if childNode.VersionConstraints != nil && childNode.VersionConstraints.String() != childNode.Version.String() {
+				item += fmt.Sprintf(" (%s)", childNode.VersionConstraints.String())
+			}
```

Comment:
What is the motivation behind this? I'd assume that version constraints will never be so long that they would worsen the readability but correct me if I'm wrong.

Reply:
It's more so to handle the case where the version constraint is specified to the exact version which would lead to a bit of redundancy– such as "mod-key' [mod-source] 1.0.0 (1.0.0). Felt that is was simpler to just omit the constraint in this case as it isn't clear that it is a version constraint. Though it might have made more sense to just append a "=" here to make the constraint more clear.

Reply:
I don't think it's a redundancy because a constraint is a different thing to the version actually installed. Relatedly, there can be many semantically equal version constraints which are represented by the same canonical format via `String()`.

For example:

 - `1.0.0`
 - `=1.0.0`
 - `= 1.0.0`

These are all equivalent but I'm not sure they would result in the same behaviour as above, which would further confuse users.

I would just keep it simple and always render it.